x-pack/plugins/fleet/cypress/integration/fleet_agent_flyout.spec.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ADD_AGENT_BUTTON, ADD_AGENT_FLYOUT } from '../screens/fleet';
+import { cleanupAgentPolicies } from '../tasks/cleanup';
+import { createAgentDoc } from '../tasks/agents';
+import { setFleetServerHost } from '../tasks/fleet';
+import { FLEET, navigateTo } from '../tasks/navigation';
+
+const FLEET_SERVER_POLICY_ID = 'fleet-server-policy';
+
+function cleanUp() {
+  cy.task('deleteDocsByQuery', {
+    index: '.fleet-agents',
+    query: { match_all: {} },
+    ignoreUnavailable: true,
+  });
+  cy.task('deleteDocsByQuery', {
+    index: '.fleet-servers',
+    query: { match_all: {} },
+    ignoreUnavailable: true,
+  });
+  cleanupAgentPolicies();
+}
+let kibanaVersion: string;
+describe('Fleet add agent flyout', () => {
+  describe('With a Fleet Server already setup', () => {
+    beforeEach(() => {
+      cleanUp();
+      let policyId: string;
+      // Create a Fleet server policy
+      cy.request({
+        method: 'POST',
+        url: '/api/fleet/agent_policies',
+        headers: { 'kbn-xsrf': 'xx' },
+        body: {
+          id: FLEET_SERVER_POLICY_ID,
+          name: 'Fleet Server policy',
+          namespace: 'default',
+          has_fleet_server: true,
+        },
+      }).then((response: any) => {
+        // setup Fleet server
+        policyId = response.body.item.id;
+      });
+
+      cy.getKibanaVersion().then((version) => {
+        kibanaVersion = version;
+      });
+
+      cy.wrap(null).then(() => {
```

Comment:
what does this `wrap(null)` do?

Reply:
it's to keep the cypress chaining while waiting on other dynamic values like `kibanaVersion` and `agentPolicy` here

Reply:
maybe we could add this trick into the cypress readme